cpu/nrf5x_common/include/nrfx.h

```diff
@@ -32,26 +32,30 @@ extern "C" {
 /**
  * @brief Enable the internal DC/DC power converter for the NRF5x MCU.
  *
- * The internal DC/DC converter is more efficient compared to the LDO regulator.
- * The downside of the DC/DC converter is that it requires an external inductor
- * to be present on the board. Enabling the DC/DC converter is guarded with
- * NRF5X_ENABLE_DCDC, this macro must be defined if the DC/DC converter is to be
- * enabled.
+ * In most cases, the internal DC/DC converter is more efficient compared to the
+ * LDO regulator. The downside of the DC/DC converter is that it requires an
+ * external LC filter to be present on the board. Per default, the DC/DC
+ * converter is enabled if an LC filter is present (VDD_LC_FILTER_REGx feature).
+ *
+ * Independent of the presence of the LC filter, the DC/DC converter(s) can be
+ * disabled by blacklisting the VDD_LC_FILTER_REGx feature, e.g. build using
+ * `FEATURES_BLACKLIST=VDD_LC_FILTER_REG1 make all`.
  */
 static inline void nrfx_dcdc_init(void)
 {
-#ifdef NRF5X_ENABLE_DCDC
-    NRF_POWER->DCDCEN = 1;
+    if (IS_ACTIVE(MODULE_VDD_LC_FILTER_REG1)) {
+        NRF_POWER->DCDCEN = 1;
+    }
 
+#ifdef POWER_MAINREGSTATUS_MAINREGSTATUS_High
     /* on CPUs that support high voltage power supply via VDDH and thus use a
-     * two stage regulator, we also enable the DC/DC converter for the first
-     * state. */
-#ifdef POWER_DCDCEN0_DCDCEN_Msk
```

Comment:
This doesn't build on microbit-v2 because `NRF_POWER_Type` has no member named `DCDCEN0`. And when using `IS_ACTIVE`, the code is always built. Maybe keep this preprocessor check (`#ifdef POWER_DCDCEN0_DCDCEN_Msk`) ?

Reply:
Yes, this could have been a rebasing mistake on my side...